src/watertap_contrib/seto/unit_models/zero_order/chemical_softening_0D.py

```diff
@@ -0,0 +1,949 @@
+        iscale.constraint_scaling_transform(self.eq_CaOH_dosing, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_in[0].flow_vol) * 3600 * 24
+        iscale.constraint_scaling_transform(self.eq_CO2_first_basin, sf)
+
+        if self.config.silica_removal and ["SiO2"] in comps:
+            sf = iscale.get_scaling_factor(self.properties_in[0].conc_mass_comp["SiO2"])
+            iscale.constraint_scaling_transform(self.MgCl2_dosing, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_in[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_recovery, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_out[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_waste_flow, sf)
+
+        comps = self.config.property_package.component_set
+        non_hardness_comps = [
+            j
+            for j in self.config.property_package.component_set
+            if j not in ["Ca_2+", "Mg_2+"]
+        ]
+
+        for c in non_hardness_comps:
+            sf = iscale.get_scaling_factor(self.properties_waste[0].flow_mass_comp[c])
+            iscale.constraint_scaling_transform(self.eq_mass_balance[c], sf)
+
+        for c in non_hardness_comps:
+            sf = iscale.get_scaling_factor(self.properties_waste[0].flow_mass_comp[c])
+            iscale.constraint_scaling_transform(self.eq_component_removal[c], sf)
+
+        sf = iscale.get_scaling_factor(self.properties_out[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_volume_mixer, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_out[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_volume_floc, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_out[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_volume_sed, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_out[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_volume_recarb, sf)
+
+        iscale.constraint_scaling_transform(self.eq_effluent_ca, 10)
+
+        iscale.constraint_scaling_transform(self.eq_effluent_mg, 10)
+
+        sf = iscale.get_scaling_factor(self.properties_in[0].conc_mass_comp["Ca_2+"])
+        iscale.constraint_scaling_transform(self.eq_mass_balance_ca, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_in[0].conc_mass_comp["Mg_2+"])
+        iscale.constraint_scaling_transform(self.eq_mass_balance_mg, sf)
+
+        sf = iscale.get_scaling_factor(self.properties_out[0].flow_vol)
+        iscale.constraint_scaling_transform(self.eq_sludge_prod, sf)
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {
+                "Feed Inlet": self.inlet,
+                "Liquid Outlet": self.outlet,
+                "Waste Outlet": self.waste,
+            },
+            time_point=time_point,
+        )
+
+    def _get_performance_contents(self, time_point=0):
+
```

Comment:
This is where you choose which variables and/or expression values would be worthwhile to show to the user. The user would run a  line such as `m.fs.chem_softening_unit.report()` and would get back values listed here. Not absolutely necessary at this stage but can be added in a subsequent PR.

Reply:
Added calculated chemical dosing variables to this function.